### PR TITLE
Closed Groups Object

### PR DESCRIPTION
**Pull request by vincentbavitz**
**PAUSED** while Maxim reworks medium groups.

### CODE REVIEW

File: ts/session/objects/ClosedGroup.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    if (
+      type === ClosedGroupType.SMALL &&
+      members.length > Constants.CLOSED_GROUP.MAX_SMALL_GROUP_MEMBERS - 1
+    ) {
+      throw new Error('Closed groups are limited to 10 memebers');
+    }
+
+    // Create Group Identity
+    const identityKeys = await libsignal.KeyHelper.generateIdentityKeyPair();
+    const id = StringView.arrayBufferToHex(identityKeys.pubKey);
```
Comment (Mikunj) [inline ts/session/objects/ClosedGroup.ts:67]: you can use `StringUtils` here

File: ts/session/objects/ClosedGroup.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      members: allMembers,
+    });
+  }
+
+  public static get(id: PubKey): ClosedGroup | undefined {
+    // Gets a closed group from its group id
+
+    const { ConversationController } = window;
+
+    const conversation = ConversationController.get(id.key) as ConversationModel;
```
Comment (Mikunj) [inline ts/session/objects/ClosedGroup.ts:145]: `conversation` could be `undefined` here

File: ts/session/objects/ClosedGroup.ts [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+export class ClosedGroup {
+  public readonly id: PubKey;
+  public readonly type: ClosedGroupType;
+  private _name: string;
+  private _admins: Array<PubKey>;
+  private _members: Array<PubKey>;
+  private _expireTimer: number;
+
+  constructor(params: ClosedGroupParams) {
+    // Setters and getters are used to prevent altering the values of name, admins, etc without a group update.
```
Comment (msgmaxim) [inline ts/session/objects/ClosedGroup.ts:43]: I don't know if that's how it should work. I don't think we should send N network requests whenever we change a field in this object. Something as expensive as this should be explicit.

File: ts/session/objects/ClosedGroup.ts [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+  recipients: Array<PubKey>;
+  members: Array<PubKey>;
+  isMediumGroup: boolean;
+  options: any;
+}
+
+export class ClosedGroup {
+  public readonly id: PubKey;
+  public readonly type: ClosedGroupType;
+  private _name: string;
```
Comment (msgmaxim) [inline ts/session/objects/ClosedGroup.ts:37]: Why the "_name" notation? I know this is sometimes done in other languages like C++, but the main reason is that `this.` part is implicit there, so `_` can at least indicate that we work with a member. I'm not sure we need this in JS.

**Pull request closed without merging**